### PR TITLE
ci: add Node.js v23

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?
CI

#### What changes did you make? (Give an overview)
Add Node.js v23 environment
#### Related Issues
No
#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

**Comment by snitin315** [comment]
Looking into the failures.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        node: [22.x, 21.x, 20.x, 18.x, "18.18.0"]
+        node: [23.x, 22.x, 21.x, 20.x, 18.x, "18.18.0"]
```
Comment (nzakas) [inline .github/workflows/ci.yml:28]: Can we remove v21.x now?

Comment (snitin315) [reply .github/workflows/ci.yml]: Yes, done.

**Comment by snitin315** [comment]
Updated deps to support Node v23

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**